### PR TITLE
doc: Add JavaDoc to BlockNetworkUtil

### PR DESCRIPTION
**Pull request by jasyohuang**

### CODE REVIEW

**Comment by pollend** [comment]
javadocs should add something to the description. This just restates the function. If this is the case then you might as well not use javadocs. Javadocs should add something that is not apparent just from the name of the function.

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 1]
```diff
@@ -26,11 +26,27 @@
     private BlockNetworkUtil() {
     }
 
+    /**
+    * getResultSide
+    * @param block - the block
```
Comment (vampcat) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: `@param block - the block`
Again, what is this supposed to mean? You're not explaining anything, you're just rewriting it. As michael mentioned, your javadoc is supposed to EXPLAIN the method, not rephrase it. Somethig like
`@param block The block that is to be rotated`

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 2]
```diff
@@ -26,11 +26,27 @@
...
+    * @param definedSide - the definedSide
```
Comment (vampcat) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: Explain this.

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 3]
```diff
@@ -26,11 +26,27 @@
...
+    * @return a side rotation of the definedSide
+    * This method's function is to get the side of a block that is rotated somehow. The rotation of the block
```
Comment (vampcat) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: The explanation goes -before- the parameters.

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 4]
```diff
@@ -26,11 +26,27 @@
...
+    * is obtained from the getRotation Method ({@link #getRotation}) and then to get the Side data, the rotation data
+    * is rotated to the defined side
+    */
     public static Side getResultSide(Block block, Side definedSide) {
         Rotation rotation = getRotation(block);
         return rotation.rotate(definedSide);
     }
 
+    /**
```
Comment (vampcat) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: See what I wrote above, and adapt it to the next 3 methods.

**Comment by pollend** [comment]
My main problem with these javadocs is they are a bit hard to follow and ramble a bit too much. The description should be concise. You really don't need a small paragraph to describe the method in question. Look at a couple of the other javadocs and they tend to have one or two sentences at most. More complex function will have longer descriptions but these methods are really not that complex.

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 5]
```diff
@@ -40,6 +53,13 @@ private static Rotation getRotation(Block block) {
         return rotation;
     }
 
+    /**
+    * Give the rotation of the block that has connections at defined sides
```
Comment (pollend) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: `get the rotated side based off the rotation of the block.`

I think this is more clear.

**Comment by jasonyonathan** [comment]
thanks ! :D

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 6]
```diff
@@ -26,11 +26,24 @@
...
+    * @param definedSide - the definedSide to define which side of the rotated block to be returned
+    * @return a side rotation of the definedSide
+    */
     public static Side getResultSide(Block block, Side definedSide) {
         Rotation rotation = getRotation(block);
         return rotation.rotate(definedSide);
     }
 
+    /**
+    * Check is the block has RotationBlockFamily and return the rotation of the block if it does
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: ~is~ if (typo/grammatical mistake)

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 7]
```diff
@@ -26,11 +26,24 @@
     private BlockNetworkUtil() {
     }
 
+    /**
+    * Get the side of a block that is rotated based on the defined side
+    *
+    * @param block - the block that is rotated
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: Remove the dash between `block` and `the block....`
Just to be consistent with Javadocs in other modules

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 8]
```diff
@@ -40,6 +53,13 @@ private static Rotation getRotation(Block block) {
         return rotation;
     }
 
+    /**
+    * Give the new side of the block based on its' rotation
+    *
+    * @param block - the block that is about to be checked its' rotation and counted its' connections
+    * @param definedSides - the definedSides on which the loop iterates the sides of the block
```
Comment (jellysnake) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: This one doesn't quite make sense. Try and re-write it without using `definedSides` in the description.

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 9]
```diff
@@ -26,11 +26,24 @@
...
+    */
     public static Side getResultSide(Block block, Side definedSide) {
         Rotation rotation = getRotation(block);
         return rotation.rotate(definedSide);
     }
 
+    /**
+    * Check is the block has RotationBlockFamily and return the rotation of the block if it does
+    *
+    * @param block - the block that is going to be checked its' rotation
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: Change the description to "the block _whose_ rotation is to be checked"

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 10]
```diff
@@ -40,6 +53,13 @@ private static Rotation getRotation(Block block) {
         return rotation;
     }
 
+    /**
+    * Give the new side of the block based on its' rotation
+    *
+    * @param block - the block that is about to be checked its' rotation and counted its' connections
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: "The block _whose_ rotation is to be checked and whose connections are to be counted"

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 11]
```diff
@@ -40,6 +53,13 @@ private static Rotation getRotation(Block block) {
...
+    * @param definedSides - the definedSides on which the loop iterates the sides of the block
+    * @return a byte of sides
+    */
     public static byte getResultConnections(Block block, byte definedSides) {
```
Comment (jellysnake) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: Overall having read this javadoc I'm still confused as to what this method actually does, the description talks about getting new sides but the method name talks about getting connections. Is the method returning sides that it's connected on?

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 12]
```diff
@@ -51,6 +71,13 @@ public static byte getResultConnections(Block block, byte definedSides) {
         return result;
     }
 
+    /**
+    * Give the new reversed side of the block based on its' rotation
+    *
+    * @param block - the block that is going to be checked
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: ~going~ (to be consistent with other Javadoc comments in this file)

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 13]
```diff
@@ -40,6 +53,13 @@ private static Rotation getRotation(Block block) {
         return rotation;
     }
 
+    /**
+    * Give the new side of the block based on its' rotation
+    *
+    * @param block - the block that is about to be checked its' rotation and counted its' connections
+    * @param definedSides - the definedSides on which the loop iterates the sides of the block
+    * @return a byte of sides
```
Comment (jellysnake) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: I'd like information on what the byte actually contains? A byte is a series of 1's and 0's. How does that relate to sides (this could point to another bit of javadoc)

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 14]
```diff
@@ -51,6 +71,13 @@ public static byte getResultConnections(Block block, byte definedSides) {
         return result;
     }
 
+    /**
+    * Give the new reversed side of the block based on its' rotation
+    *
+    * @param block - the block that is going to be checked
+    * @param connections - the connections from which the sides come from
+    * @return a byte of sides
```
Comment (jellysnake) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: See my comment about the other return

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 15]
```diff
@@ -26,11 +26,24 @@
     private BlockNetworkUtil() {
     }
 
+    /**
+    * Get the side of a block that is rotated based on the defined side
+    *
+    * @param block - the block that is rotated
+    * @param definedSide - the definedSide to define which side of the rotated block to be returned
+    * @return a side rotation of the definedSide
```
Comment (jellysnake) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: Again, confusing and doesn't actually give any extra info. You should also avoid using the name of the parameter in the description.

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 16]
```diff
@@ -40,6 +53,13 @@ private static Rotation getRotation(Block block) {
         return rotation;
     }
 
+    /**
+    * Give the new side of the block based on its' rotation
```
Comment (jellysnake) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: Slightly confusing as well. You say give the new side, but the method returns more than one side?

Comment (mjuvekar7) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: No apostrophe after "its". Change this everywhere.

**Comment by mjuvekar7** [comment]
Just one more tiny detail. I'm going to approve the GCI task though :)

**Comment by jellysnake** [comment]
Want to note that whilst github says they're outdated my comments actually are not. (At the time this is posted)

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java (hunk 13 above)
Comment (jasonyonathan) [reply src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: MichaelP said that each bit in the byte represents one side...

File: src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java [hunk 17]
```diff
@@ -40,6 +53,13 @@ private static Rotation getRotation(Block block) {
         return rotation;
     }
 
+    /**
+    * Give the new sides of the block based on its rotation
```
Comment (vampcat) [inline src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: @pollend @jellysnake What do you think about the use of the word `give` here? Can we replace it with `gives` or `returns` ?

Comment (jasonyonathan) [reply src/main/java/org/terasology/blockNetwork/BlockNetworkUtil.java]: oh yes i mean it gives...
